src/test/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistryTestData.java

```diff
@@ -38,6 +38,12 @@ public class JSONCodecRegistryTestData {
 
   public final BigDecimal NOT_EXACT_AS_INTEGER = new BigDecimal("1.25");
 
+  public final String STRING_ASCII_SAFE = "ascii-safe-string";
+
+  public final String STRING_UTF8_WITH_2BYTE_CHAR = "utf8-2-byte-\u00a2"; // cent symbol
+
+  public final String STRING_UTF8_WITH_3BYTE_CHAR = "utf8-3-byte-\u20ac"; // euro symbol
```

Comment:
question - Docs say java strings are UTF 16 https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html

and https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Character.html#unicode

both of these examples are represented with a single UTF 16 pair 

https://codepoints.net/U+00A2?lang=en
https://codepoints.net/U+20AC?lang=en

We can keep these tests, can we also add tests that require UTF16 surrogate pairs (two examples here) https://en.wikipedia.org/wiki/UTF-16#U+D800_to_U+DFFF_(surrogates)

example https://codepoints.net/U+10437?lang=en is 4 bytes in UTF-16 

then checking in the docs for the String: A String represents a string in the UTF-16 format in which supplementary characters are represented by surrogate pairs (see the section Unicode Character Representations in the Character class for more information). Index values refer to char code units, so a supplementary character uses two positions in a String.

so if you have gotten this far :) 

* is it correct to be talking about the UTF8 encoding or should it be utf16 encoding ? 
* can we also test some chars that are beyond the Basic Multilingual Plane, some docs say there are in the  "Supplementary Ideographic Plane (SIP) or the Tertiary Ideographic Plane (TIP)"  The wiki example link above has 2 chars from the SIP 
* when we have chars that require surrogte pairs, how does the charAt() work ? I read the docs and was a little confused

Reply:
Ok, yeah, I can add some comments, but more importantly will add a surrogate pair case: it is indeed 2 x UTF-16 (technically Java has "modified UCS-2" just because of details of null masking), or, 4 x UTF-8 bytes.

And as to `charAt()`: it blindly returns UTF-16 `char`, without handling surrogate pairs. But because surrogate pair characters are in `D400`-`DBFF` range, they will never be in 7-bit ASCII range, either as UTF-8 bytes UTF-16 chars. But with Java Strings we do not deal with UTF-8 bytes directly (... with JDK 9+ there is the internal optimization of using byte-based ASCII Strings... but as we saw, this details not really exposed).

Sorry, not sure above makes much sense. But:

1. Yes, I think checking of `charAt()` catches all cases we need to
2. Yes, I should still add one more check for Surrogate-pair case (4-byte)

Reply:
Wrt ^^^: added Surrogate/non-BMP unit test so we know those are caught; and added some comments on checking to note reasons why I think test is sufficient.